packages/circuit-ui/components/Tabs/Tabs.tsx

```diff
@@ -36,119 +29,76 @@ import { TabPanel } from './components/TabPanel/index.js';
 
 export interface TabsProps extends TabListProps {
   /**
-   * The index of the initially selected tab.
+   * The id of the initially selected tab.
+   *
+   * @default items[0].id
    */
-  initialSelectedIndex?: number;
+  initialSelectedId?: string;
   /**
    * A collection of tabs with an id, the tab label, and panel content.
    */
   items: {
     id: string;
-    tab: ReactNode;
+    tab: string;
     panel: ReactNode;
   }[];
 }
 
-type TabsState = {
-  selectedIndex: number;
-};
-
-export class Tabs extends Component<TabsProps, TabsState> {
-  static defaultProps = {
-    initialSelectedIndex: 0,
-  };
-
-  state = {
-    selectedIndex:
-      this.props.initialSelectedIndex || Tabs.defaultProps.initialSelectedIndex,
-  };
-
-  tabPanelsRefs = createRefs(this.props.items.length);
-
-  handleChange = (selectedIndex: number) => this.setState({ selectedIndex });
+export function Tabs({
+  items,
+  initialSelectedId = items[0].id,
+  ...props
+}: TabsProps) {
+  const [selectedId, setSelectedId] = useState(initialSelectedId);
```

Comment:
Storing the selected id instead of index is more robust in case the number or order of items changes.